processor/transformprocessor/internal/common/functions.go

```diff
@@ -30,6 +30,7 @@ func Functions[K any]() map[string]interface{} {
 		"Int":                  ottlfuncs.Int[K],
 		"ConvertCase":          ottlfuncs.ConvertCase[K],
 		"ParseJSON":            ottlfuncs.ParseJSON[K],
+		"Substring":            ottlfuncs.Substring[K],
```

Comment:
Since OTTL is separate from the transformprocessor we like to separate the function addition, which could be used in any component, and the addition of the function to the transformprocessor into separate PRs.

Reply:
Got it. I will separate it into two PRs.